src/pint/residuals.py

```diff
@@ -436,6 +437,18 @@ def chi2(self):
         assert self._chi2 is not None
         return self._chi2
 
+    @property
+    def dof(self):
+        """Return number of degrees of freedom for the DM model."""
```

Comment:
We can't split free parameters up into DM and non-DM free parameters - many (e.g. DMX) depend on both.